src/System.Text.Utf8/tests/RandomTests.cs

```diff
@@ -320,5 +321,168 @@ public void SubstringToCodeUnit(string s, char c)
                 Assert.Equal(expected, u8result.ToString());
             }
         }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("a")]
+        [InlineData("ab")]
+        [InlineData("abcdefghijklmnopqrstuvwxyz")]
+        [InlineData("ABCDEFGHIJKLMNOPQRSTUVWXYZ")]
+        [InlineData("0123456789")]
+        [InlineData(" ,.\r\n[]<>()")]
+        [InlineData("1258")]
+        [InlineData("1258Hello")]
+        [InlineData("\uABCD")]
+        [InlineData("\uABEE")]
+        [InlineData("a\uABEE")]
+        [InlineData("a\uABEEa")]
+        [InlineData("a\uABEE\uABCDa")]
+        public void CodePointEnumeratorsTests(string s)
```

Comment:
It is your call. But if the tests are that big of a hassle to write, they are already an indication (and a guid) on the quality of the API interface. Plus, much more readable.